From 844c2f1636b96204227ea74a30ac225e719c141f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:08 -0400
Subject: [powerpc] compat: 32-bit little endian machine name is ppcle, not ppc

Message-id: <1410545655-205645-380-git-send-email-dzickus@redhat.com>
Patchwork-id: 94532
O-Subject: [RHEL7 PATCH 379/626] powerpc/compat: 32-bit little endian machine name is ppcle, not ppc
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 422b9b9684db3c511e65c91842275c43f5910ae9
Author: Anton Blanchard <anton@samba.org>
Date:   Thu Mar 6 16:10:11 2014 +1100

    powerpc/compat: 32-bit little endian machine name is ppcle, not ppc

    I noticed this when testing setarch. No, we don't magically
    support a big endian userspace on a little endian kernel.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Cc: stable@vger.kernel.org # v3.10+
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/compat.h b/arch/powerpc/include/asm/compat.h
index a613d2c..b142b8e 100644
--- a/arch/powerpc/include/asm/compat.h
+++ b/arch/powerpc/include/asm/compat.h
@@ -8,7 +8,11 @@
 #include <linux/sched.h>
 
 #define COMPAT_USER_HZ  100
+#ifdef __BIG_ENDIAN__
 #define COMPAT_UTS_MACHINE "ppc\0\0"
+#else
+#define COMPAT_UTS_MACHINE "ppcle\0\0"
+#endif
 
 typedef u32  compat_size_t;
 typedef s32  compat_ssize_t;
-- 
1.7.1